File: flutter/lib/domain/use_case/deck/create_multiple_flashcards_to_deck_use_case.dart
import 'package:flashcard/data/entities/flashcard_db_entity.dart';
import 'package:flashcard/domain/repository/flashcard_repository.dart';

import '../../entities/flashcard.dart';
import '../../repository/deck_repository.dart';

class CreateMultipleFlashcardsToDeckUseCase {
  final FlashcardRepository _flashcardRepository;

  CreateMultipleFlashcardsToDeckUseCase(this._flashcardRepository);

  Future<void> call(int deckId, List<Flashcard> flashcards) async {
    if (flashcards.isEmpty) {
      throw ArgumentError('Flashcards cannot be empty');
    }
    await _flashcardRepository.addMultipleFlashcardsToDeck(deckId, flashcards);
  }
}

File: android/app/src/main/java/com/synergyboat/flutterai/domain/usecase/CreateMultipleFlashcardsToDeckUseCase.kt
package com.synergyboat.flutterai.domain.usecase

import com.synergyboat.flutterai.domain.entities.Flashcard
import com.synergyboat.flutterai.domain.repository.FlashcardRepository

class CreateMultipleFlashcardsToDeckUseCase(
    private val repository: FlashcardRepository
) {
    suspend operator fun invoke(flashcards: List<Flashcard>): List<Flashcard> {
        return flashcards.map { repository.createFlashcard(it) }
    }
}